runtime/sema/check_casting_expression.go

```diff
@@ -124,7 +124,7 @@ func (checker *Checker) VisitCastingExpression(expression *ast.CastingExpression
 						Range:        ast.NewRangeFromPositioned(leftHandExpression),
 					},
 				)
-			} else if IsSubType(leftHandType, rightHandType) {
+			} else if checker.lintEnabled && IsSubType(leftHandType, rightHandType) {
```

Comment:
Instead of having to add multiple checks, why not add a check in `checker.Hint`? We can add a new hint only if `lintEnabled` is true, and the function can be a no-op otherwise.

Reply:
The expensive part is not the reporting, but the checks for the linting. There is a follow-up issue to move the linting code out of the checker in #1269, the conditionals are only temporary to optimize the current implementation.